src/app/view/components/dot-add-persona-dialog/dot-add-persona-dialog.component.spec.ts

```diff
@@ -0,0 +1,165 @@
+@Component({
+    selector: 'dot-field-validation-message',
+    template: ''
+})
+class TestFieldValidationMessageComponent {
+    @Input() field: NgControl;
+    @Input() message: string;
+}
+
+@Component({
+    selector: 'dot-site-selector',
+    template: ''
+})
+class MockSiteSelectorComponent {
+    @Input() archive = false;
+    @Input() id = '';
+    @Input() live = true;
+    @Input() system = true;
+}
+
+describe('DotAddPersonaDialogComponent', () => {
+    let component: DotAddPersonaDialogComponent;
+    let fixture: ComponentFixture<DotAddPersonaDialogComponent>;
+    let dotDialog: DebugElement;
+    const messageServiceMock = new MockDotMessageService({
+        'modes.persona.add.persona': 'Add Persona',
+        'dot.common.dialog.accept': 'Accept',
+        'dot.common.dialog.reject': 'Cancel'
+    });
+
+    beforeEach(() => {
+        const siteServiceMock = new SiteServiceMock();
+
+        DOTTestBed.configureTestingModule({
+            declarations: [
+                DotAddPersonaDialogComponent,
+                DotCreatePersonaFormComponent,
+                TestFieldValidationMessageComponent,
+                MockSiteSelectorComponent
+            ],
+            imports: [BrowserAnimationsModule, DotDialogModule, FileUploadModule],
+            providers: [
+                DotWorkflowActionsFireService,
+                DotHttpErrorManagerService,
+                { provide: DotMessageService, useValue: messageServiceMock },
+                { provide: LoginService, useClass: LoginServiceMock },
+                { provide: SiteService, useValue: siteServiceMock }
+            ]
+        });
+
+        fixture = TestBed.createComponent(DotAddPersonaDialogComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+        dotDialog = fixture.debugElement.query(By.css('dot-dialog'));
+    });
+
+    it('should not be visible by default', () => {
+        expect(dotDialog).toBeNull();
+    });
+
+    describe('visible Dialog', () => {
+        beforeEach(() => {
+            component.visible = true;
+            fixture.detectChanges();
+            dotDialog = fixture.debugElement.query(By.css('dot-dialog'));
+        });
+
+        it('should set dialog attributes correctly', () => {
+            expect(dotDialog.componentInstance.header).toEqual('Add Persona');
+            expect(dotDialog.componentInstance.appendToBody).toBe(true);
+            expect(dotDialog.componentInstance.actions).toEqual({
+                accept: {
+                    label: 'Accept',
+                    disabled: true,
+                    action: jasmine.any(Function)
+                },
+                cancel: {
+                    label: 'Cancel',
+                    action: jasmine.any(Function)
+                }
+            });
+        });
+
+        it('should handle disable state of the accept button when form value change', () => {
+            spyOn(component, 'handlerFormValidState');
```

Comment:
You can't spy in component method, you need to test what that method does or the result of that method being called.